### PR TITLE
Feat/add pull client method

### PR DESCRIPTION
**Pull request by jpower432**
Adds the ability to pull artifacts with a pull command.

- [X] Unit tests added
- [X] E2E tests modified
- [X] Docs updated

### CODE REVIEW

**Review by sabre1041 — CHANGES_REQUESTED** [review]
This looks really good and validated the capabilities 

A few small requests for modifications

File: README.md [hunk 1]
```diff
@@ -87,7 +92,8 @@ files:
 5. Run the UOR client build command referencing the dataset config, the content directory, and optionally push to a registry location.
 
 ```
-client build --dsconfig dataset-config.yaml content-dir --push --destination localhost:5000/test/dataset:latest
+client build --dsconfig dataset-config.yaml content-dir --output my-workspace
```
[the remark this answers is not in the document]
Comment (jpower432) [reply README.md:95]: I will fix that. Thanks!

File: README.md [hunk 2]
```diff
@@ -48,6 +49,11 @@ client build my-directory --output my-workspace
 client push my-workspace localhost:5000/myartifacts:latest
 ```
 
+### Pull artifact to a location
```
Comment (sabre1041) [inline README.md:52]: Should this be renamed `artifact` -> `collection`

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**